docs/user_guide.md

```diff
@@ -0,0 +1,37 @@
+# Using S2I images
+
+## Invoking scripts embedded into an image
+
+When there is a requirement for **adding custom logic** to [S2I scripts](builder_image.md#s2i-scripts), it is possible to execute standartd scripts inside of custom scripts from user’s repository. Note that it only works when you have to add logic (for example, to generate some static assets after executing `assemble` script) and not for the case of modifying this logic. In latter case you have to replace standard scripts.
+
+To determine the location of the scripts inside of the builder image, we need to look at at the value of `io.openshift.s2i.scripts-url` label. We can use `docker inspect` for that:
+
+```console
+$ docker inspect --format=' index .Config.Labels "io.openshift.s2i.scripts-url"' openshift/wildfly-100-centos7 
+image:///usr/libexec/s2i
+```
+
+Here we are inspected `openshift/wildfly-100-centos7` builder image and find out that the scripts lie in the `/usr/libexec/s2i` directory.
+
+With this knowledge we can easily invoke any of these scripts from our own, by wrapping it's invocation.
+
+Example of `.s2i/bin/assemble` script:
+```bash
+#!/bin/bash
+echo "Before assembling"
+
+/usr/libexec/s2i/assemble
+rc=$?
+
+if [ $rc -eq 0 ]; then
+    echo "After successful assembling"
+else
+    echo "After failed assembling"
+fi
+
+exit $rc
+```
+
+Example shows custom `assemble` script that prints the message, executes standard `assemble` script from the image and prints another message depending on the exit code of the `assemble` script.
+
+In the same way, it is possible to wrap all other scripts (`save-artifacts`, `assemble-runtime` and `run`).
```

Comment:
not exactly the same.  if you wrap a run script, any logic you have after the invocation of the run script will not get executed.

also you need to make sure you "exec" the image's run script i think, otherwise your wrapping script will be pid 1 and get all the signal handling.